From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: kylechar <kylechar@chromium.org>
Date: Tue, 22 Sep 2026 07:12:16 -0700
Subject: [M153] Limit ExternalVkImageBacking usage

Original change's description:
> Limit ExternalVkImageBacking usage
>
> Don't allow creation of ExternalVkImageBacking for a client provided
> buffer outside of Fuchsia. This functionality was not intended to be
> used.
>
> Bug: 562004351
> Change-Id: If4b17f94dfde8fed1ed24f340cf13ef08cb7c7a8
> Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/8415592
> Reviewed-by: Vasiliy Telezhnikov <vasilyt@chromium.org>
> Commit-Queue: Kyle Charbonneau <kylechar@chromium.org>
> Cr-Commit-Position: refs/heads/main@{#1699830}

(cherry picked from commit 545222b904aa446dcb40a51fdca3568f0e922ddb)

Bug: 563676902,562004351
Change-Id: If4b17f94dfde8fed1ed24f340cf13ef08cb7c7a8
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/8426603
Auto-Submit: chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com <chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com>
Commit-Queue: Kyle Charbonneau <kylechar@chromium.org>
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Cr-Commit-Position: refs/branch-heads/8010@{#1617}
Cr-Branched-From: 86cee6df69e0463a839c0cc8435c9d4c259434d3-refs/heads/main@{#1681091}

diff --git a/gpu/command_buffer/service/shared_image/external_vk_image_backing_factory.cc b/gpu/command_buffer/service/shared_image/external_vk_image_backing_factory.cc
index 86935f781cd714cb9f07225a5d83e0df3f2ae74b..c7f7a20434a8e034d4e3adc6c8bf1df492ef5837 100644
--- a/gpu/command_buffer/service/shared_image/external_vk_image_backing_factory.cc
+++ b/gpu/command_buffer/service/shared_image/external_vk_image_backing_factory.cc
@@ -114,6 +114,12 @@ bool IsFormatSupported(viz::SharedImageFormat format,
       return false;
     }
   }
+
+  // Don't allow creating for a client provided GMB outside of Fuchsia. That is
+  // handled by OzoneImageBacking instead.
+  if (gmb_type != gfx::GpuMemoryBufferType::EMPTY_BUFFER) {
+    return false;
+  }
 #endif
 
   if (format.is_multi_plane()) {
